lib/qa/authorities/linked_data/find_term.rb

```diff
@@ -79,82 +80,57 @@ def expects_uri?
-            consolidated_hash[:altlabel] = altlabel unless altlabel.nil?
-            consolidated_hash[:narrower] = narrower unless narrower.nil?
-            consolidated_hash[:broader] = broader unless broader.nil?
-            consolidated_hash[:sameas] = sameas unless sameas.nil?
-            consolidated_results[uri] = consolidated_hash
-          end
-          consolidated_results.each do |res|
-            consolidated_hash = res[1]
-            consolidated_hash[:label] = sort_literals(consolidated_hash, :label)
-            consolidated_hash[:altlabel] = sort_literals(consolidated_hash, :altlabel)
-            consolidated_hash[:sort] = sort_literals(consolidated_hash, :sort)
-          end
-          consolidated_results
+        def convert_results_to_json(results)
+          json_hash = { uri: uri }
+          json_hash[:id] = results.key?(:id) && results[:id].present? ? results[:id].first.to_s : uri
+          json_hash[:label] = sort_literals(results, :label)
+          json_hash.merge!(optional_results_to_json(results))
+          predicates_hash = predicates_with_subject_uri(uri)
+          json_hash['predicates'] = predicates_hash if predicates_hash.present?
+          json_hash
         end
 
-        def sort_literals(consolidated_hash, key)
-          return nil unless consolidated_hash.key? key
-          return [] if consolidated_hash[key].blank?
-          language_sort_service.new(consolidated_hash[key], language).uniq_sorted_strings
+        def optional_results_to_json(results)
+          opt_results_json = {}
+          opt_results_json[:altlabel] = sort_literals(results, :altlabel)
+          opt_results_json[:narrower] = extract_result(results, :narrower)
+          opt_results_json[:broader] = extract_result(results, :broader)
+          opt_results_json[:sameas] = extract_result(results, :sameas)
+          opt_results_json.delete_if { |_k, v| v.blank? }
         end
 
-        def convert_term_to_json(consolidated_results)
-          json_results = []
-          consolidated_results.each do |uri, h|
-            json_hash = { uri: uri, id: h[:id], label: h[:label] }
-            json_hash[:altlabel] = h[:altlabel] unless h[:altlabel].nil?
-            json_hash[:narrower] = h[:narrower] unless h[:narrower].nil?
-            json_hash[:broader] = h[:broader] unless h[:broader].nil?
-            json_hash[:sameas] = h[:sameas] unless h[:sameas].nil?
-            json_results << json_hash
-          end
-          json_results
+        def extract_result(results, key)
+          return nil unless results.key?(key) || results[key].blank?
+          results[key]
         end
 
-        def select_json_result_for_id(json_results, id)
-          json_results.select! { |r| r[:uri].include? id } if json_results.size > 1
-          json_results.select! { |r| r[:uri].ends_with? id } if json_results.size > 1
-          json_results.first
+        def sort_literals(results, key)
+          return nil unless results.key? key
```

Comment:
This just got moved down the file and the parameter was changed from `consolidated_hash` (which no longer exists) to `results`.